multiversx_sdk_core/transaction_factories/transaction_builder.py

```diff
@@ -0,0 +1,57 @@
+from typing import List, Optional, Protocol
+
+from multiversx_sdk_core.constants import ARGS_SEPARATOR
+from multiversx_sdk_core.interfaces import IAddress
+from multiversx_sdk_core.transaction import Transaction
+
+
+class IConfig(Protocol):
+    chain_id: str
+    min_gas_limit: int
+    gas_limit_per_byte: int
+
+
+class TransactionBuilder:
+    def __init__(self,
+                 config: IConfig,
+                 sender: IAddress,
+                 receiver: IAddress,
+                 data_parts: List[str],
+                 gas_limit: int,
+                 add_data_movement_gas: bool,
```

Comment:
[optional] maybe rename this param? at least for me it's not very clear from the name what it enables.
But i also cannot think of a better name

Reply:
I've named it the same as in `js-core`, but if we think of a better name we will change it in a future PR.